openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+            else:
+                merged_std_dev = np.concatenate((other_std_dev, self_std_dev),
+                                                axis=merge_axis)
+
+            merged_tally._std_dev = np.reshape(merged_std_dev, merged_tally.shape)
+
+        # Sparsify merged tally if both tallies are sparse
+        merged_tally.sparse = self.sparse and other.sparse
+
+        return merged_tally
+
+    def to_xml_element(self):
+        """Return XML representation of the tally
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing tally data
+
+        """
+
+        element = ET.Element("tally")
+
+        # Tally ID
+        element.set("id", str(self.id))
+
+        # Optional Tally name
+        if self.name != '':
+            element.set("name", self.name)
+
+        # Optional Tally filters
+        if len(self.filters) > 0:
+            subelement = ET.SubElement(element, "filters")
+            subelement.text = ' '.join(str(f.id) for f in self.filters)
+
+        # Optional Nuclides
+        if self.nuclides:
+            subelement = ET.SubElement(element, "nuclides")
+            subelement.text = ' '.join(str(n) for n in self.nuclides)
+
+        # Scores
+        if len(self.scores) == 0:
+            msg = 'Unable to get XML for Tally ID="{0}" since it does not ' \
+                  'contain any scores'.format(self.id)
+            raise ValueError(msg)
+
+        else:
+            scores = ''
+            for score in self.scores:
+                scores += '{0} '.format(score)
+
+            subelement = ET.SubElement(element, "scores")
+            subelement.text = scores.rstrip(' ')
+
+        # Tally estimator type
+        if self.estimator is not None:
+            subelement = ET.SubElement(element, "estimator")
+            subelement.text = self.estimator
+
+        # Optional Triggers
+        for trigger in self.triggers:
+            trigger.get_trigger_xml(element)
+
+        # Optional derivatives
+        if self.derivative is not None:
+            subelement = ET.SubElement(element, "derivative")
+            subelement.text = str(self.derivative.id)
+
+        return element
+
+    def contains_filter(self, filter_type):
+        """Looks for a filter in the tally that matches a specified type
+
+        Parameters
+        ----------
+        filter_type : openmc.FilterMeta
+            Type of the filter, e.g. MeshFilter
+
+        Returns
+        -------
+        filter_found : bool
+            True if the tally contains a filter of the requested type;
+            otherwise false
+
+        """
+
+        filter_found = False
+
+        # Look through all of this Tally's Filters for the type requested
+        for test_filter in self.filters:
+            if type(test_filter) is filter_type:
+                filter_found = True
+                break
+
+        return filter_found
+
+    def find_filter(self, filter_type):
```

Comment:
Same thing here, we could just iterate, and if we find the filter, then just return the filter. If we reach the end of the loop, then just raise an error.